docs/en/week10/10-3.md

```diff
@@ -0,0 +1,310 @@
+#### Add, Norm
+
+The add norm block has two components. First is the add block, which is a residual connection, and second is the layer normalization.
+
+
+### Predictor Module
+
+The transformer predictor module follows a similar procedure as the encoder. However, there is one additional sub-block (i.e. cross-attention) to take into account. Additionally, the output of the encoder modules acts as the inputs to this module.
+
+
+
+<center>
+<img src="{{site.baseurl}}/images/week10/10-3/predictor.png" style="zoom: 100%; background-color:#DCDCDC;" /><br>
+<b>Figure 7:</b> The Predictor Module
+</center>
+
+
+#### Cross-attention
+
+The cross attention follows the query, key, and value setup used for the self-attention blocks.  However, the inputs are a little more complicated. The input to the predictor is a data point $\vect{y}\_i$, which is then passed through the self attention and add norm blocks, and finally ends up at the cross-attention block. This serves as the query for cross-attention, where the key and value pairs are the output $\vect{h}^\text{Enc}$, where this output was calculated with all past inputs $\vect{x}\_1, \cdots, \vect{x}\_{t}$.
+
+
+
+### Decoder Module
+
+Contrary to what authors of the Transformer paper define, the decoder module consists of `1D-convolution` and `Add, Norm` blocks. The output of the predictor module is fed to the decoder module and the output of the decoder module is the predicted sentence.  
+
+
+<center>
+<img src="{{site.baseurl}}/images/week10/10-3/decoder.png" style="zoom: 100%; background-color:#DCDCDC;" /><br>
+<b>Figure 8:</b> The self-attention block. The sequence of inputs is shown as a set along the 3rd dimension, and concatenated.
+</center>
+
+
+#### 1D-convolution
+
+Following this step, a 1D-convolution (aka a position-wise feed forward network) is applied. This block consists of two dense layers. Depending on what values are set, this block allows you to adjust the dimensions of the output $\vect{h}^\text{Enc}$.
+
+
+
+## Summary
+
+A set, $\vect{x}\_1$ to $\vect{x}\_{t}$ is fed through the encoder. Using self-attention and some more blocks, an output representation, $\lbrace\vect{h}^\text{Enc}\rbrace_{i=1}^t$ is obtained, which is fed to the decoder. After applying self-attention to it, cross attention is applied. In this block, the query corresponds to a representation of a symbol in the target language $\vect{y}\_i$, and the key and values are from the source language sentence ($\vect{x}\_1$ to $\vect{x}\_{t}$). Intuitively, cross attention finds which values in the input sequence are most relevant to constructing $\vect{y}\_t$, and therefore deserve the highest attention coefficients. The output of this cross attention is then fed through another 1D-convolution sub-block, and we have $\vect{h}^\text{Dec}$. For the specified target language, it is straightforward from here to see how training will commence, by comparing $\lbrace\vect{h}^\text{Dec}\rbrace_{i=1}^t$ to some target data.
```

Comment:
Why did you copy stuff from your peers? This is not cool…
Just drop a link pointing to last year's content.

Reply:
Hi, Prof. sorry to do that. We deleted the before part, but this was in middle, so kept it. We have made the new changes.